doc/tools/coqrst/coqdoc/main.py

```diff
@@ -32,10 +33,17 @@
 COQDOC_SYMBOLS = ["->", "<-", "<->", "=>", "<=", ">=", "<>", "~", "/\\", "\\/", "|-", "*", "forall", "exists"]
 COQDOC_HEADER = "".join("(** remove printing {} *)".format(s) for s in COQDOC_SYMBOLS)
 
+def tempfile():
+    fd, filename = mkstemp(prefix="coqdoc-", suffix=".v")
+    if platform.system().startswith("CYGWIN"):
+        # coqdoc doesn't accept cygwin style paths in the form "/cygdrive/c/..."
+        filename = check_output(["cygpath", "-w", filename]).strip().decode("utf-8")
```

Comment:
I'd flip the strip and the decode.  Also, you want to encode the filename before passing it, I think.

Reply:
Whitespace characters won't be changed by the decode, and they can't be part of a miultityte character.  The encode would be needed if the path to the coq source tree has utf-8 characters; so fat I think we haven't used them in any file/directory names in the source tree.

Reply:
> Whitespace characters won't be changed by the decode, and they can't be part of a miultityte character.

OK, that's fair :) I like to not reason about these things, but the argument is convincing.

> so fat I think we haven't used them in any file/directory names in the source tree.

But it's not a source tree file, right? It's wherever mkstemp creates it.  It's OK, though — python3 takes a string for that argument, so I think we're good.